clippy_lints/src/methods/collapsible_str_replace.rs

```diff
@@ -0,0 +1,160 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::get_parent_expr;
+use clippy_utils::visitors::for_each_expr;
+use clippy_utils::SpanlessEq;
+use core::ops::ControlFlow;
+use if_chain::if_chain;
+use rustc_ast::ast::LitKind;
+use rustc_errors::Applicability;
+use rustc_hir as hir;
+use rustc_hir::{ExprKind, Path, QPath};
+use rustc_lint::LateContext;
+use rustc_span::source_map::Spanned;
+use rustc_span::Span;
+
+use super::method_call;
+use super::COLLAPSIBLE_STR_REPLACE;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx hir::Expr<'tcx>, to: &'tcx hir::Expr<'tcx>) {
+    if_chain! {
+        if let Some(replace_methods) = collect_replace_calls(cx, expr, to);
+        if let Some(earliest_replace_call) = replace_methods.methods.last();
+        if let Some((_, [earliest_recv, ..], _)) = method_call(earliest_replace_call);
+        if cx.typeck_results().expr_ty(earliest_recv).peel_refs().is_str();
+
+        then {
+            // If the parent node's `to` argument is the same as the `to` argument
+            // of the last replace call in the current chain, don't lint as it would be duplicated lint
+            if let Some(parent) = get_parent_expr(cx, expr)
+                && let Some((name, [_, _, current_to], _)) = method_call(parent)
+                && name == "replace"
+                && SpanlessEq::new(cx).eq_expr(to, current_to)
+            {
+                return;
+            }
+
+            // Check if there's an earlier `str::replace` call
+            if let Some((_, [..], span_lo)) = method_call(earliest_replace_call) {
```

Comment:
`replace_methods` already contains this information. No need to check it again here.